app.js

```diff
@@ -307,16 +307,65 @@ app.post('/forgotPassword', function (req, res, next) {
 
 // Routes
 app.use('/mylibrary', require('./routes/mylibrary.js'));
-app.use('/googleapi', require('/routes/googleapi.js'));
+app.use('/googleapi', require('./routes/googleapi.js'));
+app.use('/allbooks', require('./routes/search.js'));
 
 // add book 
 app.get('/addbook', isAuthenticated, function (req, res, next) {
     var data = { title: 'Add Book' }
     res.render('addbook', data)
 })
 
-//route for books
-app.use('/allbooks', require('./routes/search.js'));
+
```

Comment:
When a book is added, a record will also need to be added to the UserBook table with the user id (from session), book id (you can get this using result.insertId after you create the book, if it already exists you'll need to retrieve it), listing date set to the current date and available set to 1. I checked the sql script and you will want to change the available column to bit so it's handled as true false versus the tinyint it was created as.

After a book is created can we flash a success message and then redirect to the my library action fo the user to see their posted book. If there is an error in adding, lets flash an error message and just render the post book page with the error message. You can still log things to the console for devs to see, especially the technical error message.

flash('success', 'This book has been added to your library')
res.redirect('/mylibrary')

flash('error', 'There was an error adding this book. Please try again later')
res.redirect('/addbook')

Reply:
there is still no error handling on this, please provide a reason why if you do not plan to add it.